### PR TITLE
Update sidebar templates

### PR DESCRIPTION
**Pull request by davidjb**
Thank you for contributing to Pipenv!

### The issue

Several URLs in the sidebar lead to broken links or outdated project URLs. Other links are http:// and insecure. And as per the changes in #3868, the top project image has an title that's not very descriptive. 

### The fix

* Make all URLs HTTPS where possible
* Update project URLs to current locations
* Make project image title consistent between sidebars

### The checklist

* [ ] Associated issue
* [ ] A news fragment in the `news/` directory to describe this fix with the extension `.bugfix`, `.feature`, `.behavior`, `.doc`. `.vendor`. or `.trivial` (this will appear in the release changelog). Use semantic line breaks and name the file after the issue number or the PR #.

### CODE REVIEW

File: docs/_templates/sidebarlogo.html [hunk 1]
```diff
@@ -50,13 +50,14 @@ <h3>Other Projects</h3>
...
-    <li><a href="http://pep8.org/">pep8.org</a></li>
-    <li><a href="http://httpbin.org/">httpbin.org</a></li>
+    <li><a href="https://pep8.org/">pep8.org</a></li>
+    <li><a href="https://httpbin.org/">httpbin.org</a></li>
     <li><a href="https://howtopython.org">HowToPython.org</a></li>
-    <li><a href="http://python-guide.org">The Python Guide</a></li>
-    <li><a href="http://python-requests.org/">Requests: HTTP for Humans</a></li>
+    <li><a href="https://python-guide.org">The Python Guide</a></li>
+    <li><a href="https://requests.readthedocs.io">Requests: HTTP for Humans</a></li>
+    <li><a href="https://github.com/kennethreitz/maya">Maya: Datetimes for Humans</a></li>
```
Comment (uranusjr) [inline docs/_templates/sidebarlogo.html:60]: Maya has transferred ownership. Also, this wasn’t present in master?

**Review by uranusjr — CHANGES_REQUESTED** [review]

File: docs/_templates/sidebarlogo.html (hunk 1 above)
Comment (davidjb) [reply docs/_templates/sidebarlogo.html]: I've updated the PR with its new location - and same with `records` as well.  This change isn't present in `master` - the latest version of the docs (https://pipenv.pypa.io/en/latest/) has the old location.

**Comment by davidjb** [comment]
Updated the PR according to @uranusjr's request.  I've also adjusted several additional URLs to use HTTPS as well, updated the `maya` location and `records` too.  That should be everything in both versions of the sidebar; let me know if anything else is required.

File: docs/_templates/sidebarlogo.html [hunk 2]
```diff
@@ -50,30 +50,31 @@ <h3>Other Projects</h3>
...
 <h3>Useful Links</h3>
 <ul>
-  <li><a href="http://docs.python-requests.org/en/latest/community/recommended/">Recommended Packages and Extensions</a></li>
+  <li><a href="https://requests.readthedocs.io/en/latest/community/recommended/">Recommended Packages and Extensions</a></li>
 
   <p></p>
 
-  <li><a href="http://github.com/pypa/pipenv">Pipenv @ GitHub</a></li>
-  <li><a href="http://pypi.org/project/pipenv">Pipenv @ PyPI</a></li>
+  <li><a href="httsp://github.com/pypa/pipenv">Pipenv @ GitHub</a></li>
```
Comment (uranusjr) [inline docs/_templates/sidebarlogo.html:73]: Typo

**Review by uranusjr — CHANGES_REQUESTED** [review]
Almost there! I’ll merge this once we get the typo fixed.

File: docs/_templates/sidebarlogo.html (hunk 2 above)
Comment (davidjb) [reply docs/_templates/sidebarlogo.html]: Ah 🤦‍♂️ Good catch!  Pushed and updated accordingly.

**Review by uranusjr — APPROVED** [review]

**Pull request merged**